metadata_backend/api/handlers.py

```diff
@@ -164,14 +164,16 @@ async def validate(self, req: Request) -> Response:
             raise web.HTTPBadRequest(reason=reason)
 
         except ParseError as error:
-            reason = f"Faulty XML file was given. Details: {error}"
-            body = json.dumps({"isValid": False, "reason": reason})
+            detail = f"Faulty XML file was given.<br/>ERROR: {error}"
+            body = json.dumps({"isValid": False, "detail": detail})
             return web.Response(body=body,
                                 content_type="application/json")
 
         except XMLSchemaValidationError as error:
-            reason = f"XML file is not valid. Details: {error}"
-            body = json.dumps({"isValid": False, "reason": reason})
+            error = str(error).replace('\n', '<br/>')  # for the frontend
```

Comment:
Same here, let's keep the original newlines. We should rather parse some information from validation error, maybe Reason and Instance. Maybe add issue about this, since this might be bigger task?

Reply:
```
Reason: invalid value 'Something wrong', it must be one of ['Whole Genome Sequencing', 'Metagenomics', 'Transcriptome Analysis', 'Resequencing', 'Epigenetics', 'Synthetic Genomics', 'Forensic or Paleo-genomics', 'Gene Regulation Study', 'Cancer Genomics', 'Population Genomics', 'RNASeq', 'Exome Sequencing', 'Pooled Clone Sequencing', 'Transcriptome Sequencing', 'Other']

Schema:

  <xs:enumeration xmlns:xs="http://www.w3.org/2001/XMLSchema" value="Whole Genome Sequencing">
    <xs:annotation>
      <xs:documentation>
         Sequencing of a single organism. 
      </xs:documentation>
    </xs:annotation>
  </xs:enumeration>

Instance:

  <STUDY_TYPE existing_study_type="Something wrong" />
```
We could try to parse necessary info from this

Reply:
Dug through the xmlschema library source code and the last commit now includes method of getting the specific outputs for Reason and Instance as shown in your example. It's commented out however, in case we wanna parse through the errors differently

Reply:
Good digging! Could you try how does that error look like in frontend?